lib/theme/main_theme.dart

```diff
@@ -0,0 +1,11 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_form_exercise/theme/colors.dart';
+import 'text_theme.dart';
+
+ThemeData mainTheme = ThemeData(
+  brightness: Brightness.light,
```

Comment:
Instead of using this `ThemeData` has a factory constructor called `light`

Reply:
Can you please elaborate on why using the ThemeData.light constructor is better than instantiating ThemeData with brightness property set to Brightness.light?
I'm not convinced to do this change because, this factory constructor has only one optional parameter (bool? useMaterial3). But I need to define more properties for the mainTheme of the app. To do so with the light constructor, I need to use copyWith() function of the ThemeData class. And again, the copyWith() function doesn't include a fontFamily parameter (which I defined as 'Inter' here). The workaround here is, to use .apply() function on the textTheme object and put provide fontFamily.
There may be some other such workarounds that I need to follow down the line. My question here is, is it worth it follow the path?

Reply:
Or correct me, if I'm diverting from the topic.

Reply:
`ThemeData` is like raw `ThemeData` but `ThemeData.light` has all the light theme details like:
Black color text, White background.
The same goes for dark themes, it has white color text, dark background.
You can control-click on it and see.